omod/src/main/java/org/openmrs/module/spa/SpaController.java

```diff
@@ -95,38 +71,28 @@ public ResponseEntity<Resource> getStaticFile(HttpServletRequest request) {
             }
 
             return ResponseEntity.ok()
-                    .cacheControl(CacheControl.maxAge(180, TimeUnit.DAYS))
-                    .headers(headers ->
-                            headers.setExpires(Instant.now().plus(180, ChronoUnit.DAYS))
-                    )
+                    .cacheControl(CacheControl.noCache().mustRevalidate())
                     .contentType(mediaType)
                     .body(resource);
         } else {
-            return ResponseEntity.notFound().build();
+            return ResponseEntity.notFound().cacheControl(CacheControl.noCache().mustRevalidate()).build();
         }
     }
 
     /**
-     * This controller method should handle everything else, and we assume that the correct response is the index.html
-     * file. Like {@link #getStaticFileWithoutCacheHeaders(String)}, this attempts to force the client to always reload
-     * this resource.
+     * This controller method should handle every other URL. We assume that the correct response is the index.html, i.e.,
+     * the single page.
```

Comment:
This comment isn't really valid anymore, is it?  Now, _this_ is the handler that is specific - targeted to handle anything with an htm, html, or /, and force all of those to just render the index.html page.  Whereas the other method is now the one that handles "every other URL", and just serves it up as requested, more or less, right?

Reply:
So, the idea is that this covers "arbitrary" URLs like `/openmrs/spa/patient/123456/chart`, whereas the other rule covers anything that looks like a file name.